### PR TITLE
Add config "install.data_disk" and remove config "install.no_data_partition"

### PR DESCRIPTION
**Pull request by johnliu55tw**
Add document for config `install.data_disk` and remove the config `install.no_data_partition`.

## Related issues
- https://github.com/harvester/harvester-installer/pull/224#issue-834211101
- https://github.com/harvester/harvester/issues/1728
- https://github.com/harvester/harvester/issues/1907

### CODE REVIEW

File: docs/install/harvester-configuration.md [hunk 1]
```diff
@@ -488,6 +488,26 @@ install:
   no_data_partition: true
 ```
 
+### `install.data_disk`
+
+#### Definition
+
+The default storage device for storing VM data.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:495]:
```suggestion
Sets the default storage device to store the VM data.
```

File: docs/install/harvester-configuration.md [hunk 2]
```diff
@@ -488,6 +488,26 @@ install:
...
+
+Default: Use the same device as [`install.device`](./#installdevice)
```
Comment (nunix) [inline docs/install/harvester-configuration.md:497]:
```suggestion
Default: Same storage device as the one set for [`install.device`](./#installdevice)
```

File: docs/install/harvester-configuration.md [hunk 3]
```diff
@@ -488,6 +488,26 @@ install:
...
+
+```yaml
+install:
+  data_disk: /dev/sdb
+```
+
+!!! note
+    If a different storage device compared to [`install.device`](./#installdevice) is provided, the
+    [`no_data_partition`](./#installno_data_partition) config will be ignored, and the additional
+    partition for storing VM data will not be created.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:509]:
```suggestion
!!! note
    If you configure a different storage device from the one set for [`install.device`](./#installdevice), the option
    [`no_data_partition`](./#installno_data_partition) will be ignored and the additional
    partition for storing VM data will not be created in the `install.device` storage device.
```

**Review by nunix — COMMENTED** [review]
Rephrased the definition, default value et the note

**Review by nunix — APPROVED** [review]

File: docs/install/harvester-configuration.md [hunk 4]
```diff
@@ -488,6 +488,26 @@ install:
   no_data_partition: true
 ```
 
+### `install.data_disk`
+
```
Comment (guangbochen) [inline docs/install/harvester-configuration.md:492]: add `_Available as of v1.0.1_`

**Comment by johnliu55tw** [comment]
@nunix please help review again due to the removal of config `install.no_data_partition`.

**Review by guangbochen — APPROVED** [review]
LGTM, and will be merged of the v1.0.1 release.

**Pull request merged**